pkg/services/ngalert/provisioning/alert_rules.go

```diff
@@ -176,7 +176,11 @@ func (service *AlertRuleService) UpdateAlertRule(ctx context.Context, rule model
 	}
 	rule.Updated = time.Now()
 	rule.ID = storedRule.ID
-	rule.IntervalSeconds, err = service.ruleStore.GetRuleGroupInterval(ctx, rule.OrgID, rule.NamespaceUID, rule.RuleGroup)
+	// we need to use the stored rule namespace uid as the folder could have
+	// been renamed, thus the new namespace uid not matching the old one and
+	// no record would be found
+	rule.IntervalSeconds, err = service.ruleStore.GetRuleGroupInterval(ctx, rule.OrgID,
+		storedRule.NamespaceUID, rule.RuleGroup)
```

Comment:
```suggestion
	rule.IntervalSeconds, err = service.ruleStore.GetRuleGroupInterval(ctx, storedRule.OrgID,
		storedRule.NamespaceUID, storedRule.RuleGroup)
```
Gets all 3 fields from `storedRule`. `rule` is user input so the other fields might not match either. For instance, if a user tries to also change the group of a rule and `rule.RuleGroup` is not what we expect, a similar bug will happen.

Reply:
Removed this whole part.